src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -51,11 +55,35 @@ object RuleUtils {
       }
     }
 
+    def isHybridScanCandidate(entry: IndexLogEntry, files: Set[FileInfo]): Boolean = {
+      // compare all the metadata of source files
+      assert(entry.relations.length == 1)
+      // TODO threshold for the number of CommonFiles
+      // TODO plan signature
```

Comment:
nit: Can you please add a bit more info for TODO to help code readability until the changes get in ultimately. Ideally, opening/linking an issue would be great.

Reply:
Done, thanks!